lib/util/sdp/index.js

```diff
@@ -437,25 +437,64 @@ function revertSimulcastForNonVP8MediaSections(localSdp, localSdpWithoutSimulcas
 }
 
 /**
- * Rewrite MSIDs in the given Unified Plan SDP with their corresponding local
- * MediaStreamTrack IDs. These IDs need not be the same.
+ * Add or rewrite MSIDs for new m= sections in the given Unified Plan SDP with their
+ * corresponding local MediaStreamTrack IDs. These can be different when previously
+ * removed MediaStreamTracks are added back (or Track IDs may not be present in the
+ * SDPs at all once browsers implement the latest WebRTC spec).
+ * @param {string} sdp
+ * @param {Map<Track.Kind, Array<Track.ID>>} trackIdsByKind
+ * @returns {string}
+ */
+function unifiedPlanAddOrRewriteNewTrackIds(sdp, trackIdsByKind) {
+  // NOTE(mmalavalli): The m= sections for the new MediaStreamTracks are usually
+  // present after the m= sections for the existing MediaStreamTracks, in order
+  // of addition.
+  const newMidsToTrackIds = Array.from(trackIdsByKind).reduce((midsToTrackIds, [kind, trackIds]) => {
+    const mediaSections = getMediaSections(sdp, kind);
+    const newMids = mediaSections.slice(mediaSections.length - trackIds.length).map(getMidForMediaSection);
```

Comment:
@syerrapragada No. Since `setLocalDescription` is not called, we can't rely on `transceiver.mid`, which is null. We have to use the fact that new m= sections are created in the order of calls to `RTCPeerConnection.addTransceiver()`.